addons/binding/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/handler/HarmonyHubHandler.java

```diff
@@ -76,6 +80,8 @@
 
     private static final int HEARTBEAT_INTERVAL = 30;
 
+    private ScheduledExecutorService buttonExecutor = Executors.newSingleThreadScheduledExecutor();
```

Comment:
You are telling me that there is a risk that one job is actually executed before another one that had been scheduled earlier already? Or is this a rather long running call, which potentially consumes all available threads in the pool? If that's the issue, it sounds ok to have a dedicated single thread to handle this.